### PR TITLE
Jinghan/implement informer API `GetCachedGroup`

### LINKED ISSUE #1008: Get metadata from informer instead of database in `PushProcessor`
Querying meatdata from database every time pushing stream feature values to offline store may cause metadata database overload. This can be avoid by querying `informer` instead.

### PR DESCRIPTION
**Pull request by jinghancc**
This PR does:
- implement informer API `GetCachedGroup`
- use informer APIs in push processor

close #1008

### CODE REVIEW

File: internal/database/metadata/informer/informer.go [hunk 1]
```diff
@@ -91,3 +93,11 @@ func (f *Informer) Cache() *Cache {
 func (f *Informer) ListCachedFeature(ctx context.Context, opt metadata.ListCachedFeatureOpt) types.FeatureList {
 	return f.Cache().Features.List(opt).Copy()
 }
+
+func (f *Informer) GetCachedGroup(ctx context.Context, groupID int) (*types.Group, error) {
+	group := f.Cache().Groups.Get(groupID).Copy()
+	if group == nil {
+		return nil, errdefs.Errorf("group %d not found", groupID)
```
Comment (lianxmfor) [inline internal/database/metadata/informer/informer.go:100]: how about?
```go
return nil, errdefs.NotFound(errdefs.Errorf("group %d not found"`, groupID)
````

**Review by wfxr — APPROVED** [review]
LGTM

File: internal/database/metadata/informer/informer.go (hunk 1 above)
Comment (jinghancc) [reply internal/database/metadata/informer/informer.go]: fixed!

File: pkg/oomstore/push_processor.go [hunk 2]
```diff
@@ -125,17 +125,14 @@ func (p *PushProcessor) Push(record types.StreamRecord) {
...
 
 func (p *PushProcessor) pushToOffline(ctx context.Context, s *OomStore, groupID int) error {
-	features, err := s.metadata.ListFeature(ctx, metadata.ListFeatureOpt{
+	features := s.metadata.ListCachedFeature(ctx, metadata.ListCachedFeatureOpt{
 		GroupID: &groupID,
 	})
-	if err != nil {
-		return err
-	}
 	entity := features[0].Entity()
```
Comment (lianxmfor) [inline pkg/oomstore/push_processor.go:132]: Is it possible for the length of `features` to be 0?

File: pkg/oomstore/push_processor.go [hunk 3]
```diff
@@ -150,15 +147,12 @@ func (p *PushProcessor) pushToOffline(ctx context.Context, s *OomStore, groupID
...
 
 func (s *OomStore) newRevisionForStream(ctx context.Context, groupID int, revision int64) error {
-	features, err := s.metadata.ListFeature(ctx, metadata.ListFeatureOpt{
+	features := s.metadata.ListCachedFeature(ctx, metadata.ListCachedFeatureOpt{
 		GroupID: &groupID,
 	})
-	if err != nil {
-		return err
-	}
 	entity := features[0].Entity()
```
Comment (lianxmfor) [inline pkg/oomstore/push_processor.go:153]: Is it possible for the length of features to be 0?

**Review by lianxmfor — APPROVED** [review]
lgtm

File: pkg/oomstore/push_processor.go (hunk 3 above)
Comment (jinghancc) [reply pkg/oomstore/push_processor.go:153]: not likely, otherwise user cannot push feature values

Comment (lianxmfor) [reply pkg/oomstore/push_processor.go:153]: lgtm!

**Pull request merged**